src/webhook/regular_user_validation.py

```diff
@@ -57,17 +53,20 @@ def get_response(req, debug=False):
     traceback.print_exc(file=sys.stdout)
     return responses.response_invalid()
 
-def is_request_allowed(username, groupname, admin_groupnames=()):
+def is_request_allowed(username, groups, admin_groupnames=()):
   """Decide if it's a special user (SA, kube:admin, etc) or not.
 
   reference: https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#request
   """
-  # Explicitly unauthenticated
+  # unauthenticated is DENY
   if username == "system:unauthenticated":
-      return False
-  if groupname in admin_groupnames:
-      return True
+    return False
+  # system users is ALLOW
   if username.startswith("kube:") or username.startswith("system:"):
+    return True
+  # SRE groups are ALLOW
+  for group in groups:
+    if group in admin_groupnames:
       return True
```

Comment:
Another way to do this that's O(N) would be
```suggestion
  if set(groups) & set(admin_groupnames):
      return True
```
but I'm guessing both lists are so small this will actually wind up being _less_ efficient...

Reply:
In case unclear, feel free to disregard this one.

Reply:
I like not having have an if in the loop.  I thought I'd seen some for logic somewhere that had if conditions but couldn't find it quickly.  I'm OK with the change just for that, but if performance is worse then will skip.  Yes they are small arrays.

Reply:
You're probably thinking of list comprehensions, something like
`[group for group in groups if group in admin_groupnames]`.
This produces a list. In this context, if the list was nonempty, it should trigger the condition.
It's close to the same thing you've got -- except that it generates the *whole* list rather than stopping on the first hit. Same thing with the `set` solution I suggested; the difference is that it only has to run through `admin_groupnames` once vs `len(groups)` times.

Anyway, this is all totally academic. What you've got is fine. (My holdup on this PR is for test coverage.)